mlflow/store/tracking/sqlalchemy_store.py

```diff
@@ -3378,6 +3399,74 @@ def _get_trace_status_from_root_span(self, spans: list[Span]) -> str | None:
                     return TraceState.OK.value
         return None
 
+    def load_spans(self, trace_id: str) -> list[Span]:
+        """
+        Load all spans for a given trace from the database.
+
+        Args:
+            trace_id: The trace ID to load spans for.
+
+        Returns:
+            List of Span objects for the given trace, ordered by start time.
+        """
+        with self.ManagedSessionMaker() as session:
+            # Query all spans for the given trace, ordered by start time
+            sql_spans = (
+                session.query(SqlSpan)
+                .filter(SqlSpan.trace_id == trace_id)
+                .order_by(SqlSpan.start_time_unix_nano)
+                .all()
+            )
+
+            if not sql_spans:
+                return []
+
+            # Convert SqlSpan objects to Span objects
+            spans = []
+            for sql_span in sql_spans:
+                span_dict = json.loads(sql_span.content)
+                span = Span.from_dict(span_dict)
+                spans.append(span)
+
+            # TODO: we should ensure the spans are logged in the correct order
```

Comment:
what happens if `load_spans` returns spans in the wrong order?

Reply:
It's sorted by start_time. I think we should ensure the spans are **logged** in correct order, but that requires more change so I added this change here to make tests pass

Reply:
what is the correct order?

Reply:
The order should be the spans order in the trace, most of the time it should be the same as start_time. But we have `mlflow.add_trace` method where if a remote trace is added, we generate a new span named `"Remote Trace <...>"` with the same start_time as the root span so the order becomes a bit tricky.
I think we should introduce a `order` field in the spans to ensure they're as the same order as when we export the whole trace, but I need to follow up this later to see if there are any potential issues